subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/observability/BuildScanFacade.kt

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.instantexecution.observability
+
+import org.gradle.api.plugins.ExtensionAware
+import org.gradle.instantexecution.InstantExecutionHost
+import org.gradle.instantexecution.extensions.unsafeLazy
+import org.gradle.internal.scan.config.BuildScanPluginApplied
+import org.gradle.kotlin.dsl.*
+import kotlin.reflect.KProperty
+
+
+internal
+interface BuildScanFacade {
+
+    var enterpriseServer: String?
+    var termsOfServiceUrl: String?
+    var termsOfServiceAgree: String?
+    var server: String?
+    var isAllowUntrustedServer: Boolean
+    var isCaptureTaskInputFiles: Boolean
+    // TODO not accessible // var isPublishAlways: Boolean
+    // TODO not accessible // var isPublishOnFailure: Boolean
+
+    fun tag(tag: String)
+
+    fun value(key: String, value: Any?)
+
+    fun instantExecutionStoreAction() {
```

Comment:
💄 I would name these methods `onInstantExecutionStoreAction`, `onInstantExecutionLoadAction` and `onInstantExecutionStats` to highlight the listener role.